### ISSUE TITLE
Misidentifying ]p

### ISSUE THREAD

**Issue by Gumnos** [heraldofsolace/VimHelpBot#48]
Created a comment with `:help ]p` and it linked to the help for `p` instead of `]p`

FWIW, this feels familiar to #47, so like that issue, it might be an old-Reddit-only issue.

**To Reproduce**

1. create a [comment in r/vim containing `:help ]p`](https://www.reddit.com/r/vim/comments/1gzuqhv/vim_challenge_for_each_line_containing_pattern/lyzvkkw/)
2. note that the [VimHelpBot links to the help for `p` instead](https://www.reddit.com/r/vim/comments/1gzuqhv/vim_challenge_for_each_line_containing_pattern/lyzvlw1/)

**Expected behavior**

VimHelpBot links to help for `]p` instead.

**Comment by Gumnos** [comment]
Adding a screenshot for clarity

![2024-11-25-193701_813x217_scrot](https://github.com/user-attachments/assets/8fa00742-5c37-4b03-b2b1-3f94eaeaf4c1)

**Comment by heraldofsolace** [comment]
Weird. It looks ok to me in both old and new Reddit. The fix for #47 replaces square brackets with fixed width square brackets which should not interfere with Markdown.

![2024-11-27T21:13:19,277604044+05:30](https://github.com/user-attachments/assets/d2a5cb7b-707c-4585-82a3-407058fd304b)

Can you do an inspect element and see the generated HTML?

**Comment by Gumnos** [comment]
On further inspection, it looks like it may be *linking* to the right place (`change.txt.html#%5Dp`) but not *displaying* the right place (just the `p`).  Viewing source on the page, I see

    <ul>
    <li><a href="https://vimhelp.org/change.txt.html#%5Dp" rel="nofollow"><code>］p</code></a> in <em>change.txt</em></li>
    </ul>

[the comments in question](https://www.reddit.com/r/vim/comments/1gzuqhv/vim_challenge_for_each_line_containing_pattern/lyzvkkw/) in case you want to poke at it too.

**Comment by heraldofsolace** [comment]
Yeah that is the correct markup. It works fine for me. Could it be that your font is missing the characters? Can you see the characters on this page? https://www.fileformat.info/info/unicode/char/ff3d/browsertest.htm

**Comment by Gumnos** [comment]
Nice troubleshooting! I copied/pasted that from Firefox's view-source into the Github issue, and it didn't render the not-a-right-square-bracket there either, so I thought it demonstrated the issue.

![2024-11-27-111221_892x261_scrot](https://github.com/user-attachments/assets/037ca985-ea60-46f3-9561-3c5ceb24e638)

But looking at the same Reddit post and the Github issue-comment above in Chromium, that glyph **does** render correctly. So there's something about Firefox that fails to render that weird bracket.
